public_html/wp-content/plugins/wordcamp-payments/includes/budget-tool.php

```diff
@@ -254,12 +262,35 @@ private static function _get_budget() {
 		return $budget;
 	}
 
+	/**
+	 * Helper function to filter out the budget items.
+	 */
```

Comment:
I'm a bit puzzled by the need for this. Why use dynamic filtering rather than just leave out the unneeded items from the declaration in `_get_default_budget()`? I'm sure there's a reason but it's not obvious.

Reply:
The `_get_default_budget` is getting bigger, it seems to me that it's not easy to read and keep up with. And I'm guessing we'll want to pull out more items from the event network down the line, so I abstract this logic into a separate function. This would facilitate the addition or removal of items as needed. If we were to directly integrate this logic into `_get_default_budget`, I feel it would make the function a bit lengthy, especially as more changes are added down the road."


```
$default_budget = array_filter($default_budget, function($item) {
    return !($item['name'] === 'tracks' || (isset($item['category']) && $item['category'] === 'speaker-event'));
});
```

If we don't add an extra filtering function, we'd just be incorporating one of the following into `_get_default_budget`. It looks okay for now, but I'm thinking it might get a bit lengthy if we add more conditions down the line. What do you think?
```
$default_budget = array_filter($default_budget, function($item) {
    if (isset($item['name']) && $item['name'] === 'tracks') {
        return false;
    }
    
    if (isset($item['category']) && $item['category'] === 'speaker-event') {
        return false;
    }

    return true;
});
```

Reply:
Or how do you feel about splitting the default budget retrieval for the next-gen network, like what's done in this [commit](https://github.com/WordPress/wordcamp.org/pull/998/commits/7c5dcc6a7687799f953318b15d0ebc68de380bbd)?